### PR TITLE
garage: fix missing stateVersion and default pkg

### PR DESCRIPTION
**Pull request by mjlbach**
###### Description of changes

* systemVersion was not in scope, added config.system.systemVersion
* garage_0_8_0 -> garage_0_8, the former does not exist

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mjlbach** [comment]
cc @raitobezarius

File: nixos/modules/services/web-servers/garage.nix [hunk 1]
```diff
@@ -60,7 +60,7 @@ in
 
     package = mkOption {
       # TODO: when 23.05 is released and if Garage 0.9 is the default, put a stateVersion check.
-      default = if versionAtLeast stateVersion "23.05" then pkgs.garage_0_8_0
+      default = if versionAtLeast config.system.stateVersion "23.05" then pkgs.garage_0_8
```
Comment (RaitoBezarius) [inline nixos/modules/services/web-servers/garage.nix:63]: No, it should use 0.8.0 and not 0.8.x.
Good catch for `stateVersion`, thank you!

Comment (RaitoBezarius) [reply nixos/modules/services/web-servers/garage.nix:63]: Ugh, 0.8.0 disappeared, hm.

Comment (RaitoBezarius) [reply nixos/modules/services/web-servers/garage.nix:63]: Okay, let's get this then.

Comment (mjlbach) [reply nixos/modules/services/web-servers/garage.nix:63]: Yeah I wasn't sure what was going on with that haha. Do you want 0_8_0 back?

**Review by RaitoBezarius — APPROVED** [review]

File: nixos/modules/services/web-servers/garage.nix (hunk 1 above)
Comment (mjlbach) [reply nixos/modules/services/web-servers/garage.nix:63]: Also I'm surprised, I thought this module was covered by tests?

Comment (RaitoBezarius) [reply nixos/modules/services/web-servers/garage.nix:63]: No no, no worries. It's just this 0.x versioning makes me confused sometimes whether 0.8 -> 0.9 is like 8.0 -> 9.0 or 0.8.0 -> 0.8.1 is 8.1 -> 8.2 which is a minor bump and all that stuff.
We are fine with 0.8.x in fact. Apologies (I'm tired :D).

Comment (RaitoBezarius) [reply nixos/modules/services/web-servers/garage.nix:63]: > Also I'm surprised, I thought this module was covered by tests?

Nix is lazy, stateVersion is unchecked then

**Pull request merged**